### PR TITLE
add functions to apply Givens rotation

### PR DESCRIPTION
**Pull request by davydden**
a part of https://github.com/dealii/dealii/pull/7222/

### CODE REVIEW

File: include/deal.II/lac/lapack_full_matrix.h [hunk 1]
```diff
@@ -175,6 +175,22 @@ class LAPACKFullMatrix : public TransposeTable<number>
   void
   rank1_update(const number a, const Vector<number> &v);
 
+  /**
+   * Apply <a href="https://en.wikipedia.org/wiki/Givens_rotation">Givens</a>
+   * rotation @p csr (a triplet of cosine, sine and radius, see
```
Comment (bangerth) [inline include/deal.II/lac/lapack_full_matrix.h]: move the `</a>` *after* the word "rotation"

File: include/deal.II/lac/lapack_full_matrix.h [hunk 2]
```diff
@@ -175,6 +175,22 @@ class LAPACKFullMatrix : public TransposeTable<number>
...
+   * Utilities::LinearAlgebra::givens_rotation())
+   * to this matrix in the plane spanned by the @p i'th and @p k'th unit vectors.
+   * If @p left is <code>true</code>, the rotation is applied from left
+   * and only rows @p i and @p k are affected.
+   * Otherwise, transpose of the rotation matrix is applied from right
+   * and only columns @p i and @p k are affected.
```
Comment (bangerth) [inline include/deal.II/lac/lapack_full_matrix.h]: do you mean to say that it is *also* applied from the right, i.e., from the left *and* right? Or *only* from the right? Might be worth clarifying.

File: include/deal.II/lac/vector.h [hunk 3]
```diff
@@ -298,6 +298,17 @@ class Vector : public Subscriptor
   void
   grow_or_shrink(const size_type N);
 
+  /**
+   * Apply <a href="https://en.wikipedia.org/wiki/Givens_rotation">Givens</a>
+   * rotation @p csr (a triplet of cosine, sine and radius, see
```
Comment (bangerth) [inline include/deal.II/lac/vector.h]: same here

**Comment by davydden** [comment]
@bangerth i updated the documentation, hopefully it's more clear now.

**Comment by davydden** [comment]
I think testers were not through (strangely enough), it showed green only from 1 test... Noticed that happens after force-push, but let's hope nothing is broken...

**Comment by bangerth** [comment]
Hm, it was all green at the time when I merged...

**Pull request merged**